### PR TITLE
T345689 build three versions plain

### PR DESCRIPTION
**Pull request by rti**
In this PR:
- refactor the github action and build script to work similarily for every component (wikibase, wdqs, ...)
- build wikibase (base) and wikibase-bundle with the same make target
- combine github actions so that everything (lint, build, test, publish to GHCR) happens in one workflow

### CODE REVIEW

File: .github/actions/prep-env-and-cache/action.yml
Comment (adeeri) [inline .github/actions/prep-env-and-cache/action.yml:1]: Do these `action.yml` files need to be in separate sub-directories?

File: channels/lts.env [hunk 1]
```diff
@@ -1,34 +1,45 @@
...
+# Head of master on 2023/04/15 as per https://github.com/magnusmanske/quickstatements/commits/master
 QUICKSTATEMENTS_COMMIT_HASH=a05157d54a387636a73613b60a3119616882d749
-# Head of master on 01/03/2023
+# Head of master on 01/03/2023 (last commit is from 2023-01-17 as per https://bitbucket.org/magnusmanske/magnustools/src/master/)
 MAGNUSTOOLS_COMMIT_HASH=51a32efcfb7113ac682b8d9a6c30106f5ec106e9
-# 1.0.2 latest release on 01/03/2023 (main branch README states 1.39 compatability)
+# 1.0.2 latest release on 04/10/2022 (main branch README states 1.39 compatability) https://github.com/ProfessionalWiki/WikibaseLocalMedia/tree/1.0.2#platform-requirements
 WIKIBASELOCALMEDIA_COMMIT_HASH=66fd4e622d9e39125a68fe67f49c9657a0aa35dc
-# 2.0.1 latest release on 01/03/2023 (main branch README states 1.39 compatability)
+# 2.0.1 latest release on 26/03/2022 (main branch README states 1.39 compatability) https://github.com/ProfessionalWiki/WikibaseEdtf/blob/master/README.md#installation
```
Comment (RickiJay-WMDE) [inline channels/lts.env:20]: Please use a consistent date format - 2023/04/15 and 04/10/2022 are both used here

File: Makefile [hunk 2]
```diff
@@ -23,7 +23,7 @@ ifdef SUITE
 ifndef GITHUB_ACTIONS
 	-@make lint
 endif
-	@bash test/scripts/test_suite.sh ${SUITE}
+	@bash test/scripts/test_suite.sh ${SUITE} ${CHANNEL}
```
Comment (RickiJay-WMDE) [inline Makefile:26]: Please update readme to demonstrate how to locally test

File: .github/actions/prep-env-and-cache/action.yml
Comment (rti) [reply .github/actions/prep-env-and-cache/action.yml:1]: Aparently yes :(

```


 
--
Can't find 'action.yml', 'action.yaml' or 'Dockerfile' under '/home/runner/work/wikibase-release-pipeline/wikibase-release-pipeline/.github/actions/prep-env-and-cache.yml'. Did you forget to run actions/checkout before running your local action?

https://github.com/wmde/wikibase-release-pipeline/actions/runs/6324278278/job/17173523750

File: channels/lts.env (hunk 1 above)
Comment (rti) [reply channels/lts.env]: Good catch, thanks, done.

File: Makefile (hunk 2 above)
Comment (rti) [reply Makefile:26]: thanks, done

File: README.md
Comment (RickiJay-WMDE) [inline README.md:1]: Please update Build Commands as well!

**Review by RickiJay-WMDE — CHANGES_REQUESTED** [review]

File: README.md
Comment (rti) [reply README.md:1]: In the README that is? Done. Good catch, thanks!

**Review by RickiJay-WMDE — DISMISSED** [review]

File: test/docker-compose.yml [hunk 3]
```diff
@@ -32,7 +32,11 @@ services:
       - MW_ADMIN_PASS
       - MW_SCRIPT_PATH
 
-      - WMDE_RELEASE_VERSION
+      - WIKIBASE_SUITE_RELEASE_MAJOR_VERSION
+      - WIKIBASE_SUITE_RELEASE_MINOR_VERSION
+      - WIKIBASE_SUITE_RELEASE_PATCH_VERSION
+      - WIKIBASE_SUITE_RELEASE_PRERELEASE_VERSION
+
```
Comment (lorenjohnson) [inline test/docker-compose.yml:39]: Remove?

File: versions.env [hunk 4]
```diff
@@ -0,0 +1,51 @@
+# shellcheck disable=2034 # unused variable
```
Comment (lorenjohnson) [inline versions.env:1]: Can we must merge this into `variables.env`?

**Review by lorenjohnson — CHANGES_REQUESTED** [review]
LGTM except the two comments.

File: versions.env (hunk 4 above)
Comment (rti) [reply versions.env]: yes, done

File: test/docker-compose.yml (hunk 3 above)
Comment (rti) [reply test/docker-compose.yml]: yes, done

File: .github/workflows/build_and_test.yml [hunk 5]
```diff
@@ -610,53 +210,103 @@ jobs:
...
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ inputs.github_token }}
+
+      - name: Store built docker image on GHCR
+        if: github.ref == 'refs/heads/main'
+        uses: wmde/tag-push-ghcr-action@v3
+        with:
+          image_name: "wikibase/${{ inputs.name }}"
+          tag: ${{ github.run_id }}
```
Comment (lorenjohnson) [inline .github/workflows/build_and_test.yml:312]: Really happy about the clean-up and simplification here!

**Review by lorenjohnson — APPROVED** [review]
Great! Will merge and then do some additional testing of the Upgrade test.

**Pull request merged**